packages/fxa-settings/src/components/DropDownAvatarMenu/index.tsx

```diff
@@ -0,0 +1,124 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import React, { useState } from 'react';
+import { gql, useMutation } from '@apollo/client';
+import Avatar from '../Avatar';
+import AlertBar from '../AlertBar';
+import { useAccount } from '../../models';
+import sentryMetrics from 'fxa-shared/lib/sentry';
+import { clearSignedInAccountUid } from '../../lib/cache';
+import { useClickOutsideEffect, useBooleanState } from 'fxa-react/lib/hooks';
+import { useEscKeydownEffect } from '../../lib/hooks';
+import { ReactComponent as SignOut } from './sign-out.svg';
+
+export const DESTROY_SESSION_MUTATION = gql`
+  mutation destroySession($input: DestroySessionInput!) {
+    destroySession(input: $input) {
+      clientMutationId
+    }
+  }
+`;
+
+export const DropDownAvatarMenu = () => {
+  const { displayName, primaryEmail } = useAccount();
+  const [isRevealed, setRevealed] = useState(false);
+  const toggleRevealed = () => setRevealed(!isRevealed);
+  const avatarMenuInsideRef = useClickOutsideEffect<HTMLDivElement>(
+    setRevealed
+  );
+  useEscKeydownEffect(setRevealed);
+  const [alertBarRevealed, revealAlertBar, hideAlertBar] = useBooleanState();
+  const dropDownId = 'drop-down-avatar-menu';
+
+  // TODO: FXA-6249
+  const [destroySession, { data, error }] = useMutation(
+    DESTROY_SESSION_MUTATION,
+    {
+      onError: (error) => {
+        revealAlertBar();
+        sentryMetrics.captureException(error);
+      },
+    }
+  );
+
+  const signOut = () => {
+    destroySession({
+      variables: { input: {} },
+    });
+  };
+
+  if (data) {
+    clearSignedInAccountUid();
+    window.location.assign(`${window.location.origin}/signin`);
+    return null;
+  }
```

Comment:
@dannycoates it feels like I'm missing something here, but I'm not entirely sure what.

In the content server, we `clearSignedInAccountUid` (removing account uid from `localStorage`), clear everything set on the `Session` object created in the content server, then navigate back to `signin` with `queryString` params. Here I'm removing the account uid from `localStorage`, but I don't _think_ we have anything else to clear, and, I don't think we need to transfer params since when the user goes to hit `beta/settings` they're redirected to `get_flow`?

Reply:
this redirect url seems fine